examples/api/lib/widgets/implicit_animations/animated_value.0.dart

```diff
@@ -0,0 +1,371 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/material.dart';
+
+/// Flutter code sample for [AnimatedValue].
+
+void main() => runApp(const AnimatedValueApp());
+
+/// Prizes come in all shapes!
+enum Prize implements ValueKey<Prize> {
+  circle,
+  triangle,
+  square,
+  hexagon;
+
+  @override
+  Prize get value => this;
+
+  @override
+  String toString() => name;
+}
+
+class PrizeSpinner extends StatelessWidget {
+  const PrizeSpinner({super.key});
+
+  // This build method returns an AnimatedRotation, which is a subtype
+  // of AnimatedValue!
+  @override
+  AnimatedValue<double> build(BuildContext context) {
+    final Prize? prize = FreePrizes.maybeOf(context);
+    final double turns = (prize?.index ?? math.Random().nextDouble()) / 4 + 1 / 8;
+
+    // Changing a widget's key will cause its state to reset.
+    // So if an AnimatedValue widget animates from an initialValue,
+    // changing the key will make the animation run again!
+
+    return AnimatedRotation(
+      key: prize,
+      initialValue: prize != null ? turns - 2 : null,
+      turns: turns,
+      duration: const Duration(seconds: 2),
+      curve: Curves.easeOutCubic,
+      onEnd: () {
+        if (prize != null) {
+          FreePrizes.recolor(context);
+
+          showDialog<void>(
+            context: context,
+            builder: (BuildContext context) => PrizePopup(prize: prize),
+          );
+        }
+      },
+      child: const Align(
+        alignment: Alignment.centerRight,
+        child: Icon(
+          Icons.arrow_right_alt_rounded,
+          size: 125,
+          color: Color(0xff202020),
+        ),
+      ),
+    );
+  }
+}
+
+abstract interface class _PrizeScope {
+  /// Randomize the prize!
+  void randomize();
+
+  /// Color the screen!
+  void recolor();
+}
+
+class FreePrizes extends InheritedWidget {
+  const FreePrizes({super.key, this.prize, required super.child});
+
+  final Prize? prize;
+
+  static Prize? maybeOf(BuildContext context) {
+    return context.dependOnInheritedWidgetOfExactType<FreePrizes>()?.prize;
+  }
+
+  static _PrizeScope _prizeScope(BuildContext context) {
+    return context.findAncestorStateOfType<_AnimatedValueAppState>()!;
```

Comment:
This structure is a little odd. You are inserting an inherited widget to get quick access to `prize`, but then you need to walk the widget tree here to access the thing that injected the `prize`? Maybe the inherited widget should give you direct access to both.